shm/map.cpp

```diff
@@ -0,0 +1,43 @@
+#include "map.hpp"
+#include <map>
+#include <random>
+#include <string>
+
+Map::Map() {
+
+    allIslands.reserve(NUM_OF_ISLANDS);
+
+    std::map<std::string, int> repsChecker;
+
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> distrib(MAP_RND_DISTRIBUTION_MIN, MAP_RND_DISTRIBUTION_MAX);
+
+    bool coordinateExist = false;
+    int32_t positionX = 0;
+    int32_t positionY = 0;
+    std::string coordinateXY = "";
+
+    for (std::size_t i = 0; i < NUM_OF_ISLANDS; ++i) {
+        do {
+            positionX = distrib(gen);
+            positionY = distrib(gen);
+
+            coordinateXY = std::string(std::to_string(positionX) + std::to_string(positionY));
+            auto tryInsert = repsChecker.emplace(coordinateXY, 0);
+            coordinateExist = tryInsert.second;
+
+        } while (!coordinateExist);
+
+        allIslands.emplace_back(Island(Coordinates(positionX, positionY)));
```

Comment:
emplace_back will call directly constructor. When you create temporary object better use push_back.